plugins/woocommerce/client/blocks/assets/js/base/utils/get-product-data.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import {
+	Store as WooCommerce,
+	ProductData,
+	SelectedAttributes,
+} from '@woocommerce/stores/woocommerce/cart';
+import { store } from '@wordpress/interactivity';
+
+/**
+ * Internal dependencies
+ */
+import {
+	AvailableVariation,
+	getMatchedVariation,
+} from './variations/get-matched-variation';
+
+// Stores are locked to prevent 3PD usage until the API is stable.
+const universalLock =
+	'I acknowledge that using a private store means my plugin will inevitably break on the next store release.';
+
+const { state: wooState } = store< WooCommerce >(
+	'woocommerce',
+	{},
+	{ lock: universalLock }
+);
+
+export const getProductData = (
+	id: number,
+	productType: string,
+	availableVariations?: AvailableVariation[],
+	selectedAttributes?: SelectedAttributes[]
+): ( ProductData & { id: number } ) | null => {
+	let productId = id;
+	let productData: ProductData | undefined;
+
+	// If both the condition satisfies that's a variable product, we need to get the selected variation data.
```

Comment:
What do you think about rewriting this comment as:

```suggestion
	// If `availableVariations` and `selectedAttributes` are defined, it means it's a variable product and we need to get the selected variation data.
```